crates/miden-testing/tests/agglayer/bridge_in.rs

```diff
@@ -0,0 +1,206 @@
+extern crate alloc;
+
+use core::slice;
+
+use miden_lib::account::faucets::NetworkFungibleFaucet;
+use miden_lib::account::wallets::BasicWallet;
+use miden_lib::agglayer::{agglayer_faucet_component, bridge_out_component, create_claim_note};
+use miden_lib::note::WellKnownNote;
+use miden_objects::account::{
+    Account,
+    AccountStorageMode,
+    AccountType,
+    StorageSlot,
+    StorageSlotName,
+};
+use miden_objects::asset::{Asset, FungibleAsset, TokenSymbol};
+use miden_objects::note::{
+    Note,
+    NoteAssets,
+    NoteExecutionHint,
+    NoteInputs,
+    NoteMetadata,
+    NoteRecipient,
+    NoteTag,
+    NoteType,
+};
+use miden_objects::transaction::OutputNote;
+use miden_objects::{Felt, FieldElement, Word};
+use miden_testing::{AccountState, Auth, MockChain};
+use rand::Rng;
+
+/// Tests the bridge-in flow: CLAIM note -> Aggfaucet (FPI to Bridge) -> P2ID note created.
+#[tokio::test]
+async fn test_bridge_in_claim_to_p2id() -> anyhow::Result<()> {
+    let mut builder = MockChain::builder();
+
+    // CREATE BRIDGE ACCOUNT (with bridge_out component for MMR validation)
+    // --------------------------------------------------------------------------------------------
+    let bridge_storage_slot_name = StorageSlotName::new("miden::agglayer::bridge").unwrap();
+    let bridge_storage_slots = vec![StorageSlot::with_empty_map(bridge_storage_slot_name)];
+    let bridge_component = bridge_out_component(bridge_storage_slots);
+    let bridge_account_builder = Account::builder(builder.rng_mut().random())
+        .storage_mode(AccountStorageMode::Public)
+        .with_component(bridge_component);
+    let bridge_account = builder.add_account_from_builder(
+        Auth::IncrNonce,
+        bridge_account_builder,
+        AccountState::Exists,
+    )?;
```

Comment:
would be great to have a util function to create all bridge components in one go